### PR TITLE
Fixing assert on variable live range for simd12 LCL / STORE_LCL

### LINKED ISSUE #78898: Test failure  JIT\\Regression\\JitBlue\\Runtime_63354\\Runtime_63354\\Runtime_63354.cmd
Run: [runtime-coreclr jitstress 20221126.2](https://dev.azure.com/dnceng-public/public/_build/results?buildId=94284&view=ms.vss-test-web.build-test-results-tab&runId=1987692&resultId=119602&paneView=debug)

Failed test:
```
coreclr windows x86 Checked jitstress1 @ Windows.10.Amd64.Open

- JIT\\Regression\\JitBlue\\Runtime_63354\\Runtime_63354\\Runtime_63354.cmd

coreclr windows x86 Checked jitstress1_tiered @ Windows.10.Amd64.Open

- JIT\\Regression\\JitBlue\\Runtime_63354\\Runtime_63354\\Runtime_63354.cmd

coreclr windows x86 Checked jitstress2 @ Windows.10.Amd64.Open

- JIT\\Regression\\JitBlue\\Runtime_63354\\Runtime_63354\\Runtime_63354.cmd

coreclr windows x86 Checked jitstress2_tiered @ Windows.10.Amd64.Open

- JIT\\Regression\\JitBlue\\Runtime_63354\\Runtime_63354\\Runtime_63354.cmd
```
**Error message:**
```
Assert failure(PID 2576 [0x00000a10], Thread: 5528 [0x1598]): Assertion failed '!m_VariableLiveRanges->back().m_EndEmitLocation.Valid()' in 'Runtime_63354:Test2(System.Numerics.Vector3):System.Numerics.Vector3' during 'Generate code' (IL size 52; hash 0x3273ce14; FullOpts)

File: D:\a\_work\1\s\src\coreclr\jit\codegencommon.cpp Line: 8795
Image: C:\h\w\9B6408B0\p\corerun.exe
Return code:      1
Raw output file:      C:\h\w\9B6408B0\w\A9090942\uploads\Reports\JIT.Regression\JitBlue\Runtime_63354\Runtime_63354\Runtime_63354.output.txt
Raw output:
BEGIN EXECUTION
"C:\h\w\9B6408B0\p\corerun.exe" -p "System.Reflection.Metadata.MetadataUpdater.IsSupported=false"  Runtime_63354.dll
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=C:\h\w\9B6408B0\p
C:\h\w\9B6408B0\w\A9090942\e\JIT\Regression\JitBlue\Runtime_63354\Runtime_63354\Runtime_63354.cmd
Expected: True
Actual:   False


Stack trace
   at JIT_Regression._JitBlue_Runtime_63354_Runtime_63354_Runtime_63354_._JitBlue_Runtime_63354_Runtime_63354_Runtime_63354_cmd()
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
```

### PR DESCRIPTION
**Pull request by BrianBohe**
Intend to fix [78898](https://github.com/dotnet/runtime/issues/78898).

Op GT_STORE_LCL_VAR for target type SIMD12 always stores the result on the stack when FEATURE_SIMD is on. Liveness and the variable descriptor were updated after emitting this operation. This PR updates the tree's variable descriptor and liveness.

Having said that, LSRA assigned a target register for GT_STORE_LCL_VAR is assigned a target register in LSRA although the result (v01 arg0) is stored on the stack. For that reason, the tree target reg is being updated inside our specific case. LSRA already has the information about the available registers, then might be better to add this SIMD12 information there and avoid changing the tree after. I am opening this PR to see if this is triggering other problems.

The same behavior may be happening for GT_STOREIND for the same target type but I cannot get an example.

### CODE REVIEW

**Comment by BruceForstall** [comment]
I wonder why the arm64 `genStoreLclTypeSIMD12` code supports reg/reg `mov` but the x64 code doesn't.

File: src/coreclr/jit/simdcodegenxarch.cpp [hunk 1]
```diff
@@ -938,10 +938,10 @@ void CodeGen::genStoreLclTypeSIMD12(GenTree* treeNode)
 {
     assert((treeNode->OperGet() == GT_STORE_LCL_FLD) || (treeNode->OperGet() == GT_STORE_LCL_VAR));
 
+    treeNode->SetRegNum(REG_STK); // we always store the result on stack
```
Comment (BruceForstall) [inline src/coreclr/jit/simdcodegenxarch.cpp:941]: It seems odd to call this. Isn't the problem that LSRA shouldn't be assigning a register to this node at all?

**Review by BruceForstall — CHANGES_REQUESTED** [review]

**Comment by kunalspathak** [comment]
`genStoreLclTypeSIMD12()` was just missing a case for a scenario where `targetReg` (the destination register for `STORE_LCL_VAR`) is not `REG_NA`. In this case, it is `xmm1` and for such, we should just do the move and no need to store the variable on stack. Of course, if we are doing a `mov`, we need to also `genProduceReg()` which will update the life of `V01`.

**Comment by kunalspathak** [comment]
Pushed fixes at other places where we were not tracking the liveness of `treeNode` during `genCodeForStoreLclVar()` and `genCodeForStoreLclFld()` on both x64 and arm64.

**Comment by kunalspathak** [comment]
If jitstress looks ok, we should also kick off gcstress @BrianBohe

**Review by BruceForstall — APPROVED** [review]
LGTM.

Are there any asm diffs? It would be nice to summarize interesting examples here.

**Comment by kunalspathak** [comment]
> LGTM.
> 
> Are there any asm diffs? It would be nice to summarize interesting examples here.

The only difference is in coreclr_tests for the test case that was failing.

image

Here is the diff which removes the extra store on stack.

image

**Comment by BrianBohe** [comment]
Errors are CI infrastructure related (timeouts) or unrelated

**Pull request merged**